x-pack/plugins/spaces/public/legacy_urls/redirect_legacy_url.ts

```diff
@@ -20,14 +20,15 @@ export function createRedirectLegacyUrl(
 ): SpacesApiUi['redirectLegacyUrl'] {
   return async function ({
     path,
+    useToast,
     aliasPurpose,
     objectNoun = DEFAULT_OBJECT_NOUN,
   }: RedirectLegacyUrlParams) {
     const [{ notifications, application }] = await getStartServices();
     const { currentAppId$, navigateToApp } = application;
-    const appId = await currentAppId$.pipe(first()).toPromise(); // retrieve the most recent value from the BehaviorSubject
+    const appId = await firstValueFrom(currentAppId$); // retrieve the most recent value from the BehaviorSubject
 
-    if (aliasPurpose === 'savedObjectConversion') {
+    if (aliasPurpose === 'savedObjectConversion' || useToast) {
```

Comment:
I would prefer to remove aliasPurpose entirely in favour of useToast, but this shows how it could work in conjunction if we want to keep aliasPurpose